pypdf/generic/_data_structures.py

```diff
@@ -995,26 +983,26 @@ def __init__(
         # The inner list has two elements:
         #  Element 0: List
         #  Element 1: str
-        self.operations: List[Tuple[Any, Any]] = []
+        self._operations: List[Tuple[Any, Any]] = []
 
         # stream may be a StreamObject or an ArrayObject containing
         # multiple StreamObjects to be cat'd together.
-        if stream is not None:
+        if stream is None:
+            super().set_data(b"")
+        else:
             stream = stream.get_object()
             if isinstance(stream, ArrayObject):
-                data = b""
+                data = bytearray()
```

Comment:
I just ran the benchmark with all of your changes, but replaced this back with `b""`. It was faster. So the `bytearray()` seems to rather slow it down.

Reply:
Without the bytearray:

* Text Extraction Speed: 2.2s
* Image Extraction Speed: 2.7s
* Watermarking Speed: 0.4s

Reply:
Alright, I reverted that